src/witan/workspace_executor/core.clj

```diff
@@ -431,146 +438,95 @@
   (when-let [from-node (get name->node (first (:from node [])))]
-    [(.create!
-      (LinearRouter.
-       (meld-name (:name from-node) (:name node))
-       from-node
-       node
-       (atom nil)))]))
+    (LinearRouter.
+     (meld-name (:name from-node) (:name node))
+     from-node
+     node
+     (atom nil))))
 
 (defmethod create-router :from-many
   [name->node node]
-  ;move the linear router creation into here i reckon, can bin off the protocol then i reckon.
-  (.create!
-   (MergeRouter.
-    (map 
-     #(get name->node %)
-     (:from node))
-    node
-    (atom nil))))
+  (let [from-nodes (map 
+                    #(get name->node %)
+                    (:from node))
+        merge-nodes (map #(map->Node
+                     {:inbound (async/chan)
+                      :name (keyword (str (name (:name %)) "_merge"))})
+                   from-nodes)
+        linear-routers (map #(LinearRouter.
+                              (meld-name (:name %1) (:name node)) %1 %2 (atom nil))
+                            from-nodes
+                            merge-nodes)]
+    (cons (MergeRouter. (meld-merge-name node) linear-routers merge-nodes node)
+          linear-routers)))
 
 (defn create-predicate-router
   [name->node edge-name->router node]
   (let [replay-routers (map (partial edge->router edge-name->router) 
                             (gather-replay-edges name->node node))
         pred-replay-fn (predicate-replay replay-routers)
         from-node (get name->node (first (:from node)))]
-    [(.create!
-       (LinearRouter.
-        (meld-name (:name from-node) (:name node))
-        from-node
-        node
-        (atom nil)))
-     (.create!
-      (PredicateRouter.
-       (:name node)
-       (map (partial get name->node) (:to node))
-       node
-       pred-replay-fn
-       replay-routers))]))
+    [(LinearRouter.
+      (meld-name (:name from-node) (:name node))
+      from-node
+      node
+      (atom nil))
+     (PredicateRouter.
+      (:name node)
+      (map (partial get name->node) (:to node))
+      node
+      pred-replay-fn
+      replay-routers)]))
 
 (defmulti create-invoker 
-  (fn [workspace ^Node node] (:type node)))
+  (fn [^Node node func] (:type node)))
 
 (defmethod create-invoker :predicate
-  [workspace ^Node node]
-  (pipe!
-   (create-invoker-of-type
-    PredicateInvoker
-    workspace
-    (:name node)
-    node)))
+  [^Node node func]
+  (PredicateInvoker.
+   node
+   (async/chan)
+   func))
 
 (defmethod create-invoker :default
-  [workspace ^Node node]
-  (pipe!
-   (create-invoker-of-type
-    Invoker
-    workspace
-    (:name node)
-    node)))
+  [^Node node func]
+  (Invoker.
+   node
+   (async/chan)
+   func))
 
 (defn reduce-nodes-to-map-name->node
   [nodes]
   (reduce (fn [a n] (assoc a (:name n) n)) {} nodes))
 
+(def map-flat (comp flatten keep))
+
+(defn activate-all!
+  [activatables]
+  (doseq [a activatables]
+    (activate! a)))
+
 (s/defn build!
```

Comment:
well it depends if you like to reflect bangs up. This is a bang function because it calls bangers